### PR TITLE
Move test-history code into the test-infra repository.

### PR DESCRIPTION
**Pull request by spxtr**
~~Waiting on https://github.com/kubernetes/test-infra/pull/3~~ ready to go

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml [hunk 1]
```diff
@@ -6,15 +6,15 @@
         - timed: 'H 0 * * *'
     scm:
         - git:
-            url: https://www.github.com/kubernetes/kubernetes
```
Comment (rmmh) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml]: don't you mean kubernetes/test-infra?

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml [hunk 2]
```diff
@@ -6,15 +6,15 @@
...
             branches:
                 - master
             browser: githubweb
-            browser-url: https://github.com/kubernetes/kubernetes
+            browser-url: https://github.com/kubernetes/test-history
             skip-tag: true
     builders:
         - shell: |
-            cd hack/jenkins/test-history
+            cd jenkins/test-history
```
Comment (rmmh) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml]: in the current state, it should be `cd jenkins/test-history`

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-test-history.yaml]: Done.

**Comment by spxtr** [comment]
@rmmh PTAL, the prereq is merged.

**Comment by spxtr** [comment]
Waiting on https://github.com/kubernetes/test-infra/pull/8

**Comment by rmmh** [comment]
LGTM

**Pull request merged**